src/php/lib/Grpc/Interceptor.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/*
+ *
+ * Copyright 2017 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+namespace Grpc;
+
+class InterceptorHelper{
+  public static function intercept($channel, Interceptor $interceptor){
```

Comment:
How about moving this function to the `Channel` class itself?

`Channel::intercept($channel, $interceptor)`

Reads nicer too and one fewer class.

Also, can we possibly be able to accept one or multiple interceptors passed to this function like:

`InterceptorHelper::intercept($channel, $i1, $i2, $i3)`

That would be nice?

Reply:
How about move it into Grpc\InterceptorChannel::intercept().
Grpc\Channel is implemented by C code and add a method in that parts needs to add code in a lot of places.